src/components/Menu/Menu.test.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from "react";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { axe } from "jest-axe";
+
+import Menu, { ListItemsData } from "./Menu";
+
+const defaultListItems: ListItemsData[] = [
+  {
+    type: "action",
+    id: "item-title-1",
+    label: "I'm item 1",
+    onClick: () => {
+      console.log("Item Title 1 clicked");
+    },
+  },
+  {
+    type: "action",
+    id: "item-title-2",
+    label: "I'm item 2",
+    onClick: () => {
+      console.log("Item Title 2 clicked");
+    },
+  },
+  {
+    type: "action",
+    id: "item-title-3",
+    label: "I'm item 3",
+    onClick: () => {
+      console.log("Item Title 3 clicked");
+    },
+  },
+];
+
+describe("Menu Accessibility", () => {
+  it("passes axe accessibility test", async () => {
+    const { container } = render(
+      <Menu labelText={"Menu"} listItemsData={defaultListItems} />
+    );
+    expect(await axe(container)).toHaveNoViolations();
+  });
+});
+
+describe("Menu opens and closes", () => {
+  it("renders content when it is opened", async () => {
+    render(<Menu labelText={"Menu"} listItemsData={defaultListItems} />);
+    const openButton = screen.getByText("Menu").closest("button");
+    expect(openButton).toBeInTheDocument();
+    openButton.click();
```

Comment:
In the next test, there's isn't an `await waitFor` call. If you use fireEvent.click, I think you can remove the await here.